interfaces/builtin/gpg_keys.go

```diff
@@ -49,6 +49,15 @@ deny @{HOME}/.gnupg/trustdb.gpg w,
 # trust to access the user's keys and the level of trust for the random_seed
 # is commensurate with accessing the private keys.
 owner @{HOME}/.gnupg/random_seed wk,
+
+# allow access to gpg-agent's extra socket for passphrase entry.
+# This socket exposes a very limited subset of gpg-agent's functionality and
+# only allows the confined application to use gpg's cryptographic functions
+# but not anything related to managing the keys (adding, trusting etc).
+owner /run/user/[0-9]*/gnupg/S.gpg-agent.extra wr,
```

Comment:
Absolutely stunning elaboration, thank you @jdstrand. I agree that a separate interface is likely the way to go if full access is something that is desirable.

To start an agent specific to the snap is likely a good solution if no external interfaces (besides the keyboard) are required. My concern here is stuff like Smartcards (```pcscd```, ``scdaemon``). This is pretty much the same story again, only now we can't run two scdaemons concurrently, as they compete for the same device. I'll have to research this scenario a little more though...

> All considered and since the existing interfaces are expected to work with a snap-specific agent, I'd prefer this scenario be properly documented[...]

I feel that's a good solution for a decent chunk of interface users. This could maybe even be implemented as a snapcraft extension.